### PR TITLE
replace action shellcheck.yml with differential_shellcheck.yml

### PR DESCRIPTION
**Pull request by nhosoi**
- replace action shellcheck.yml with differential_shellcheck.yml
- add shellcheck_roles.yml to group_vars to manage roles containing shell scripts.

### CODE REVIEW

File: playbooks/templates/.github/workflows/differential-shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Repository checkout
+        uses: {{ gha_checkout_action }}
+        with:
+          fetch-depth: 0
+
+      - id: ShellCheck
+        name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (richm) [inline playbooks/templates/.github/workflows/differential-shellcheck.yml:29]:
```suggestion
{%- raw %}
          token: ${{ secrets.GITHUB_TOKEN }}
```

File: playbooks/templates/.github/workflows/differential-shellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - if: ${{ always() }}
+        name: Upload artifact with ShellCheck defects in SARIF format
+        uses: actions/upload-artifact@v3
+        with:
+          name: Differential ShellCheck SARIF
+          path: ${{ steps.ShellCheck.outputs.sarif }}
```
Comment (richm) [inline playbooks/templates/.github/workflows/differential-shellcheck.yml:36]:
```suggestion
          path: ${{ steps.ShellCheck.outputs.sarif }}
{%- endraw +%}
```

File: playbooks/templates/.github/workflows/differential-shellcheck.yml (hunk 1 above)
Comment (richm) [reply playbooks/templates/.github/workflows/differential-shellcheck.yml:29]: without the `raw`, jinja will try to process the `{{ secrets.GITHUB_TOKEN }}` and fail

**Comment by richm** [comment]
Also `git rm -f playbooks/templates/.github/workflows/shellcheck.yml`

**Comment by nhosoi** [comment]
> Also `git rm -f playbooks/templates/.github/workflows/shellcheck.yml`

I was wondering about the file. Since we put it in the `absent_files`, can we expect it will be automatically absent?
https://github.com/linux-system-roles/.github/pull/13/files#diff-ee877de6bbd0a00a6b5cad8bdb7ec1f27fbb4d5d96e5e77b27a56fce75562273R12-R14

How about the individual roles which do not have shell scripts? Do we want to do `git rm shellcheck.yml` and merge it, too?

**Comment by richm** [comment]
> > Also `git rm -f playbooks/templates/.github/workflows/shellcheck.yml`
> 
> I was wondering about the file. Since we put it in the `absent_files`, can we expect it will be automatically absent? 

It will be removed from the role repos, but it also should be removed from this `.github` repo as well. 

> https://github.com/linux-system-roles/.github/pull/13/files#diff-ee877de6bbd0a00a6b5cad8bdb7ec1f27fbb4d5d96e5e77b27a56fce75562273R12-R14
> 
> How about the individual roles which do not have shell scripts? Do we want to do `git rm shellcheck.yml` and merge it, too?

Yes.

**Review by richm — APPROVED** [review]

**Comment by richm** [comment]
@nhosoi please also revert this PR

**Comment by nhosoi** [comment]
> @nhosoi please also revert this PR

Thank you, @richm. I've opened https://github.com/linux-system-roles/.github/pull/18.

**Pull request merged**